src/main/java/com/zendesk/maxwell/schema/SchemaStore.java

```diff
@@ -229,8 +266,79 @@ public static SchemaStore restore(Connection connection, MaxwellContext context)
 		return s;
 	}
 
-	private void restoreFrom(Connection conn, BinlogPosition targetPosition) throws SQLException, InvalidSchemaError {
-		PreparedStatement p;
+	private List<ResolvedSchemaChange> parseDeltas(String json) throws IOException {
+		if ( json == null )
+			return null;
+
+		return mapper.readerFor(listOfResolvedSchemaChangeType).readValue(json.getBytes());
+	}
+
+	private HashMap<Long, HashMap<String, Object>> buildSchemaMap(Connection conn) throws SQLException {
+		HashMap<Long, HashMap<String, Object>> schemas = new HashMap<>();
+
+		PreparedStatement p = conn.prepareStatement("SELECT * from `schemas` where server_id = ?");
```

Comment:
what I'm thinking: store the first_schema_id (I'd call them first_schema_id and parent_schema_id, base_schema_id for me might be understood as first or parent... not an english speaker though =)) in `saveDerivedSchema` (get it from parent.first_schema_id), and in buildSchemaMap the query changes to `"SELECT * from `schemas` where server_id = ? and first_schema_id = ?"`.

That way it is easy to only select the schemas that are part of this schema's chain.

If it won't be expensive, ignore the comment =)